lookout/core/helpers/server.py

```diff
@@ -0,0 +1,160 @@
+"""Utils to work with lookout-sdk binary."""
+import io
+import json
+import logging
+import os
+import pathlib
+import random
+from shutil import copyfileobj
+import socket
+import subprocess
+import sys
+import tarfile
+import tempfile
+from typing import Optional
+from urllib.error import HTTPError
+from urllib.request import urlopen
+
+from lookout.core.api.version import __version__ as binver
+
+
+class LookoutSDK:
+    """
+    Wrapper class for lookout-sdk binary.
+
+    Allows you to query analyzers the same way lookout server do.
+    About lookout-sdk read https://github.com/src-d/lookout-sdk
+    """
+
+    _log = logging.getLogger("LookoutSDK")
+
+    def __init__(self, analyzer_port: Optional[int]=None):
+        """
+        Lookout SDK initialization.
+
+        Fetches lookout-sdk executable if it is missing.
+        :param analyzer_port: Running analyzer port on localhost. Finds available port if not \
+            specified. It is user responsibility to run analyzer on this port.
+        """
+        self._version = binver
+        self._exefile = (pathlib.Path(tempfile.gettempdir()) /
+                         "lookout-sdk-ml" / ("lookout-sdk-%s" % self._version))
+        if not self._exefile.exists():
+            self.fetch()
+        self._port = find_port() if analyzer_port is None else analyzer_port
```

Comment:
The port should be scanned inside `_run()` and never cached. The reason is that it can be taken by an external party at any time.

Reply:
We should find a port for Analyzer and then run it. So... I think what I can do.

Reply:
You can check that the port is still available inside `_run(), and if it doesn't , repeat the scanning

Reply:
ok, let's Join this class and https://github.com/src-d/style-analyzer/blob/1bc68e0e68e1a27dd2b325a490a936b9a81d9c5e/lookout/style/format/benchmarks/analyzer_context_manager.py?